src/borg/archiver.py

```diff
@@ -541,9 +543,10 @@ def create_inner(archive, cache, fso):
                     except (FileNotFoundError, PermissionError) as e:
                         self.print_error('Failed to execute command: %s', e)
                         return self.exit_code
-                    pipe = proc.stdout
+                    pipe_bin = proc.stdout
                 else:  # args.paths_from_stdin == True
-                    pipe = sys.stdin
+                    pipe_bin = sys.stdin.buffer
+                pipe = TextIOWrapper(pipe_bin, encoding='utf-8', errors='surrogateescape')
```

Comment:
utf-8? do we know?

how about keeping it as bytes?

maybe also check what the internal recursion is working with, iirc it is using byte paths, not text paths.

Reply:
I checked and it's using strings (i traced the type of `path` and `name` arguments to `Archiver._process_any`). But in any case `surrogateescape` makes it safe: if there is any non-utf8 byte in the input it will be encoded as a lone surrogate (a python implementation trick) and then this will be reverted whenever interacting with the filesystem. This is all documented in [pep 383](https://www.python.org/dev/peps/pep-0383/).

Maybe i shouldn't have passed the encoding tho (to use the default system-guessed encoding). I'm removing that.

Reply:
ah, ok.